f5lbaasdriver/test/tempest/tests/scenario/test_l7policies_and_rules.py

```diff
@@ -0,0 +1,176 @@
+        self._delete_l7policy(self.l7policy.get('id'))
+        self._wait_for_load_balancer_status(self.load_balancer.get('id'))
+        self._run_traffic('server1', headers={'X-HEADER': 'yes'})
+        self._remove_members()
+
+    def test_policy_redirect_pool_cookie_contains(self):
+        '''Redirect to pool when cookie contains value.'''
+
+        self.l7policy = self._create_l7policy(**self.redirect_pool_args)
+        rule_args = {'type': 'COOKIE', 'compare_type': 'CONTAINS',
+                     'key': 'cook', 'value': 'es'}
+        self._create_l7rule(self.l7policy.get('id'), **rule_args)
+        self._run_traffic('server1')
+        self._run_traffic('server2', cookies={'cook': 'test'})
+        self._delete_l7policy(self.l7policy.get('id'))
+        self._wait_for_load_balancer_status(self.load_balancer.get('id'))
+        self._run_traffic('server1', cookies={'cook': 'test'})
+        self._remove_members()
+
+
+class TestL7PolicyBasicRedirectToUrl(TestL7PolicyBasic):
+    '''Testing L7 Redirect To Pool Policies deployed to BIG-IP device.'''
+
+    def setUp(self):
+        super(TestL7PolicyBasicRedirectToUrl, self).setUp()
+        self._redirect_url_args()
+
+    def _redirect_url_args(self):
+        self.redirect_url_args = {
+            'listener_id': self.listener.get('id'),
+            'admin_state_up': 'true', 'action': 'REDIRECT_TO_URL',
+            'redirect_url': 'http://10.190.0.20'}
+
+    def test_policy_redirect_url_hostname_ends_with(self):
+        '''Redirect traffic to url when hostname ends with value.'''
+
+        self.l7policy = self._create_l7policy(**self.redirect_url_args)
+        rule_args = {'type': 'HOST_NAME', 'compare_type': 'ENDS_WITH',
+                     'value': '1.1.1'}
+        self._create_l7rule(self.l7policy.get('id'), **rule_args)
+        self._run_traffic('server1')
+        self._run_traffic('coverity-doc', headers={'Host': '11.1.1.1'})
+        self._run_traffic('coverity-doc', headers={'Host': '1.1.1.1'})
+        self._run_traffic('server1', headers={'Host': '10.1.32.2'})
+        self._delete_l7policy(self.l7policy.get('id'))
+        self._wait_for_load_balancer_status(self.load_balancer.get('id'))
+        self._run_traffic('server1', headers={'Host': '1.1.1.1'})
+        self._remove_members()
+
+
+class TestL7PolicyBasicReject(TestL7PolicyBasic):
+    '''Testing L7 Redirect To Pool Policies deployed to BIG-IP device.'''
+
+    def setUp(self):
+        super(TestL7PolicyBasicReject, self).setUp()
+        self._reject_args()
+
+    def _reject_args(self):
+        self.reject_args = {
+            'listener_id': self.listener.get('id'),
+            'admin_state_up': 'true', 'action': 'REJECT'}
+
+    def test_policy_reject_header_starts_with(self):
+        '''Reject traffic when header value starts with value.'''
+
+        self.l7policy = self._create_l7policy(**self.reject_args)
+        rule_args = {'type': 'HEADER', 'compare_type': 'CONTAINS',
```

Comment:
Paul, the test name indicate that it wants to check compare type of "starts with", but in the rule args, compare_type is set to "CONTAINS"